miner/worker.go

```diff
@@ -908,6 +909,15 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 	return false
 }
 
+func (w *worker) fetchPendingL1Messages() []types.L1MessageTx {
+	includedL1BlockNumber := rawdb.ReadLastL1MessageInL2Block(w.eth.ChainDb(), w.chain.CurrentHeader().Hash())
+	lastBlockNumberToFetch := rawdb.ReadSyncedL1BlockNumber(w.eth.ChainDb())
+	if *includedL1BlockNumber+w.chainConfig.L1Config.NumL1MessagesPerBlock < *lastBlockNumberToFetch {
+		*lastBlockNumberToFetch = *includedL1BlockNumber + w.chainConfig.L1Config.NumL1MessagesPerBlock
+	}
+	return rawdb.ReadL1MessagesInRange(w.eth.ChainDb(), *includedL1BlockNumber+1, *lastBlockNumberToFetch, true)
+}
```

Comment:
L1 messages are indexed by QueueIndex, not L1 block number, so you cannot use `ReadSyncedL1BlockNumber` here.

```suggestion
func (w *worker) fetchPendingL1Messages() []types.L1MessageTx {
	includedL1BlockNumber := rawdb.ReadLastL1MessageInL2Block(w.eth.ChainDb(), w.chain.CurrentHeader().Hash())
	// TODO: handle nil
	first := *includedL1BlockNumber + 1
	last := first + w.chainConfig.L1Config.NumL1MessagesPerBlock - 1
	return rawdb.ReadL1MessagesInRange(w.eth.ChainDb(), first, last, false)
}
```